### PR TITLE
fix(db): 500 on hitting the `balance/` endpoint

### PR DESCRIPTION
**Pull request by Alex6323**

### CODE REVIEW

File: src/db/collections/outputs/mod.rs [hunk 1]
```diff
@@ -320,8 +320,8 @@ impl MongoDb {
                 .unwrap_or_default();
 
             Ok(Some(BalancesResult {
-                total_balance: balances.total_balance.amount as u64,
-                sig_locked_balance: balances.sig_locked_balance.amount as u64,
+                total_balance: balances.total_balance[0].amount as u64,
```
Comment (Alexandcoats) [inline src/db/collections/outputs/mod.rs:323]: You should be able to remove the `Balances` struct, and instead directly deserialize a `BalancesResult`, by using `$first` in a `$project`.

Comment (Alex6323) [reply src/db/collections/outputs/mod.rs]: Thanks, Alex. I bet you're right on this. Will try to do it a bit later.

File: src/db/collections/outputs/mod.rs [hunk 2]
```diff
@@ -81,11 +81,11 @@ pub struct OutputWithMetadataResult {
     pub metadata: OutputMetadataResult,
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default, Deserialize)]
 #[allow(missing_docs)]
 pub struct BalancesResult {
-    pub total_balance: u64,
-    pub sig_locked_balance: u64,
+    pub total_balance: f64,
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:87]: Change this to `String` (see comment below)

**Review by grtlr — CHANGES_REQUESTED** [review]

**Review by grtlr — CHANGES_REQUESTED** [review]
Does the current query properly handle outputs that are spent within the same milestone? @Alexandcoats could you please also verify that?

**Comment by Alex6323** [comment]
If it's spent in the same milestone then `{ "metadata.spent_metadata.spent": null },` will not match?

**Comment by grtlr** [comment]
Yeah, you're right, I must have been confused by the different versions of the query.

**Review by Alexandcoats — DISMISSED** [review]

**Comment by Alex6323** [comment]
I'll approve as well :trollface: since @Alexandcoats made the query much simpler and possibly more performant as well.

**Review by grtlr — APPROVED** [review]

**Pull request merged**